### PR TITLE
Brings back collector power scaling with the amount of plasma

### PR DESCRIPTION
**Pull request by ACCount12**
This PR makes the efficiency of radiation collectors increase with plasma pressure and decrease with plasma temperature once again. This feature was removed in the recent radiation rework, and, honestly, it doesn't have anything to do with recent radiation rework.

This time, the power scaling has coefficients in defines. For example, pressure scaling coefficient is 0.15, and this means that filling plasma tanks up to 1013kPa increases power output 135% instead of old 335%. This makes power scaling less overpowered and makes increasing power generation by other means (core gas mix, emitter setup) more viable.

I plan to add more gas interactions to the collectors in the future, but this is out of scope of this PR.

:cl: ACCount
balance: Radiation collector power scaling is back. Have fun with high-pressure supercooled plasma!
/:cl:

### CODE REVIEW

**Comment by Jalleo** [comment]
Please add a changelog for the PR.

**Comment by KorPhaeron** [comment]
@ninjanomnom

File: code/modules/power/singularity/collector.dm
Comment (ninjanomnom) [inline code/modules/power/singularity/collector.dm]: these shouldn't be vars on the collector if you're recalculating every process

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
You should graph changes like this if you haven't. You've made increasing the temperature increase the power output here. https://www.desmos.com/calculator/qr3wfutaer

A max bonus of ~50% would be more in line with the effort involved. Probably less tbh.

I'll touch on this again though when I go for the more serious pass on balancing power output. For now a 50% possible boost and a slight reduction in power output is good.

File: code/modules/power/singularity/collector.dm
Comment (duncathan) [reply code/modules/power/singularity/collector.dm]: it shouldn't be possible; there are checks elsewhere in gas mixtures for this

**Comment by ACCount12** [comment]
I checked the changes [here](http://grimalschi.github.io/calque/index.html), I just typed the `temperature_ratio` wrong.

The "135%" I listed is not bonus, it's total. It means 135% of old output, i.e. +35%. And max cap of +50% for everything, including supercooling plasma to TCMB? Nah, I'll pass on your suggestion.

**Comment by ninjanomnom** [comment]
The current power output is right about at enough for standard setups. Adding a bonus to power with no reduction at all just allows setups that shouldn't be viable. Like for example setups that are too safe such as pure n2.

**Comment by ninjanomnom** [comment]
Though admittedly I still need to tweak it to be about right for that particular case, I think pure n2 build are technically barely viable for station power. Part of that though is difference in station size which I was holding off for dealing with in a bigger power balance pass.

**Comment by ACCount12** [comment]
Who are you to decide what core setups should be viable and what should not? If setups that are "too safe" shouldn't exist, go remove solars.

Changelog added, ready to merge.

**Comment by optimumtact** [comment]
He's allowed to decide that because core maintainers trust him, so you can be assured when ninja speaks about this, he speaks with the authority of kor as our headcoder and design lead

**Comment by optimumtact** [comment]
lose the hostile attitude to people and try to work with them instead or you are going to find yourself on a one way trip

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
What I said above

**Comment by ACCount12** [comment]
Not going to cap it at 50%. Feel free to tweak the coefficients, but +50% from TCMB plasma is just pointless. Why even bother?

**Comment by ACCount12** [comment]
I feel like I'm getting balance memed here, and I hate that feeling. Like I said, +50% from TCMB is retardation.

**Comment by Dorsisdwarf** [comment]
Plasma in collectors should slowly heat up, so that if you want to run a supercharged engine you have to swap them out frequently.

**Comment by ACCount12** [comment]
Sadly, cooperating with balance memelords has proven to be impossible. Usually the best course of action is to ignore them, but that stops working the moment someone gives them maintainer rights.

**Comment by ACCount12** [comment]
I kind of like the "heating up" idea tho, it's not retarded nerf-it-to-the-ground approach and I see how this might work. Need to do some testing with current radiation levels, so don't expect an update quick.

**Comment by ninjanomnom** [comment]
We had a conversation in irc and you were fine with the changes talked about there, I was just waiting for you to implement the changes.

Also the heating up part is something I suggested as a possibility then kek

**Comment by ACCount12** [comment]
I wasn't fine because you went "75% max" and that was bullshit.

**Comment by ninjanomnom** [comment]
50% max is more than enough, people don't need to have the ability to double their power output because they chilled a tank of plasma.

It was another change you were talking about that'd make it more okay to raise that since it was more involved but I forget what that was atm.

**Comment by ninjanomnom** [comment]
Also as I said a soft cap is just as fine as a hard cap, just the normal practical max should be ~50%

**Comment by optimumtact** [comment]
if you don't want to work with us you can always leave, it's no skin of our nose.

**Comment by lordpidey** [comment]
Honestly, rather than worry about the pressure/temperature of the gas, simply look at the number of moles in it.

**Comment by ACCount12** [comment]
That's how it worked before, but @ninjanomnom wasn't happy with that kind of scaling and silently removed it in his radiation PR.

**Comment by lordpidey** [comment]
I mean still do the whole non-linear scaling you have going on here.

Have X moles produce Y power, have 2x moles produce 1.25Y power, etc....

**Comment by ninjanomnom** [comment]
It certainly wasn't silent, it was in the changelog and I discussed it in coderbus many times.

**Comment by optimumtact** [comment]
since you seem to have no interest in working with ninja on this

**Comment by optimumtact** [comment]
feel free to ping for a reopen or make a new pr if you feel ready to work cooperatively

**Comment by ACCount12** [comment]
I have no interest in nerfing things into the ground, that's all.

**Pull request closed without merging**
